agent/consul/stream/event.go

```diff
@@ -0,0 +1,16 @@
+package stream
+
+// FilterObject returns the object in the event to use for boolean
+// expression filtering.
+func (e *Event) FilterObject() interface{} {
+	if e == nil || e.Payload == nil {
+		return nil
+	}
+
+	switch e.Payload.(type) {
+	case *Event_ServiceHealth:
+		return e.GetServiceHealth().ServiceNode
```

Comment:
🤔 is this equivalent to the other RPC endpoint? I thought we allowed filtering on Node and Check status etc. too there so we'd need to return the whole CheckServiceNode I think to be equivalent.

Reply:
If we don't already have tests for filters in the current HTTP blocking endpoint test, perhaps we should add some to verify that switching the internals out still produce the same result?

Reply:
ServiceNode actually is the whole CheckServiceNode, but i'll rename it since that's confusing and it was just a name picked early on.